libsql-storage-server/src/fdb_store.rs

```diff
@@ -0,0 +1,153 @@
+use crate::store::FrameData;
+use crate::store::FrameStore;
+use async_trait::async_trait;
+use foundationdb::api::NetworkAutoStop;
+use foundationdb::tuple::pack;
+use foundationdb::tuple::unpack;
+use foundationdb::Transaction;
+use tracing::error;
+
+pub struct FDBFrameStore {
+    _network: NetworkAutoStop,
+}
+
+impl FDBFrameStore {
+    pub fn new() -> Self {
+        let _network = unsafe { foundationdb::boot() };
+        Self { _network }
+    }
+
+    async fn get_max_frame_no(&self, txn: &Transaction, namespace: &str) -> u64 {
+        let max_frame_key = format!("{}/max_frame_no", namespace);
+        let result = txn.get(&max_frame_key.as_bytes(), false).await;
+        if let Err(e) = result {
+            error!("get failed: {:?}", e);
+            return 0;
+        }
+        if let Ok(None) = result {
+            error!("page not found");
+            return 0;
+        }
+        let frame_no: u64 = unpack(&result.unwrap().unwrap()).expect("failed to decode u64");
+        tracing::info!("max_frame_no ({}) = {}", max_frame_key, frame_no);
+        frame_no
+    }
+
+    async fn insert_with_tx(
+        &self,
+        namespace: &str,
+        txn: &Transaction,
+        frame_no: u64,
+        frame: FrameData,
+    ) {
+        let frame_data_key = format!("{}/f/{}/f", namespace, frame_no);
+        let frame_page_key = format!("{}/f/{}/p", namespace, frame_no);
+        let page_key = format!("{}/p/{}", namespace, frame.page_no);
+
+        txn.set(&frame_data_key.as_bytes(), &frame.data);
+        txn.set(&frame_page_key.as_bytes(), &pack(&frame.page_no));
+        txn.set(&page_key.as_bytes(), &pack(&frame_no));
+    }
+}
+
+#[async_trait]
+impl FrameStore for FDBFrameStore {
+    async fn insert_frame(&self, namespace: &str, page_no: u32, frame: bytes::Bytes) -> u64 {
```

Comment:
why do we need both `insert_frame` and `insert_frames`?

Reply:
Right, both are not needed. I will remove `insert_frame` in a follow up PR.